modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/index.d.ts

```diff
@@ -0,0 +1,213 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+export function FrontendDataSet({
+	actionParameterName,
+	activeViewSettings,
+	apiURL,
+	appURL,
+	bulkActions,
+	creationMenu,
+	currentURL,
+	customDataRenderers,
+	customViewsEnabled,
+	filters,
+	formId,
+	formName,
+	id,
+	initialSelectedItemsValues,
+	inlineAddingSettings,
+	inlineEditingSettings,
+	items,
+	itemsActions,
+	namespace,
+	nestedItemsKey,
+	nestedItemsReferenceKey,
+	onActionDropdownItemClick,
+	onBulkActionItemClick,
+	overrideEmptyResultView,
+	pagination,
+	portletId,
+	selectedItemsKey,
+	selectionType,
+	showManagementBar,
+	showPagination,
+	showSearch,
+	sidePanelId,
+	sorting,
+	style,
+	views,
+}: IFrontendDataSetProps): JSX.Element;
+
+interface IFrontendDataSetProps {
+	actionParameterName?: string;
+	activeViewSettings?: string;
+	apiURL?: string;
+	appURL?: string;
+	bulkActions?: any[];
+	creationMenu?: {
+		primaryItems?: any[];
+		secondaryItems?: any[];
+	};
+	currentURL?: string;
+	customDataRenderers?: any;
+	customViewsEnabled?: boolean;
+	enableInlineAddModeSetting?: {
+		defaultBodyContent?: object;
+	};
+	filters?: (
+		| IAutocompleteFilter
+		| ICustomFilter
+		| ICheckboxFilter
+		| IDateRangeFilter
+		| IRadioFilter
+	)[];
+	formId?: string;
+	formName?: string;
+	id: string;
+	initialSelectedItemsValues?: any[];
+	inlineAddingSettings?: {
+		apiURL: string;
+		defaultBodyContent: object;
+	};
+	inlineEditingSettings?: [
+		boolean,
+		{
+			alwaysOn: boolean;
+			defaultBodyContent: object;
+		}
+	];
+	items?: any[];
+	itemsActions?: TItemsActions[];
+	namespace?: string;
+	nestedItemsKey?: string;
+	nestedItemsReferenceKey?: string;
+	onActionDropdownItemClick?: any;
+	onBulkActionItemClick?: any;
+	overrideEmptyResultView?: boolean;
+	pagination?: {
+		deltas?: [
+			{
+				href?: string;
+				label: number;
+			}
+		];
+		initialDelta: number;
+	};
+	portletId?: string;
+	selectedItemsKey?: string;
+	selectionType?: ['single', 'multiple'];
+	showManagementBar?: boolean;
+	showPagination?: boolean;
+	showSearch?: boolean;
+	sidePanelId?: string;
+	sorting?: [
+		{
+			direction?: ['asc', 'desc'];
+			key?: string;
+		}
+	];
+	style?: ['default', 'fluid', 'stacked'];
+	views: [
+		{
+			component?: any;
+			contentRenderer?: string;
+			contentRendererModuleURL?: string;
+			label?: string;
+			name?: string;
+			schema?: object;
+			thumbnail?: string;
+		}
+	];
+}
+
+type TItemsActions = {
+	data?: {
+		confirmationMessage?: string;
+		id?: string;
+		method?: 'delete' | 'get';
+		permissionKey?: string;
+	};
+	href?: string;
+	icon?: string;
+	label?: string;
+	target?: 'async' | 'headless' | 'link' | 'modal' | 'sidePanel';
+};
+
+type TFilters = {
+	id: string;
+	label: string;
+	type:
+		| 'autocomplete'
+		| 'checkbox'
+		| 'dateRange'
+		| 'number'
+		| 'radio'
+		| 'text';
```

Comment:
I updated [docs](https://liferay.atlassian.net/wiki/spaces/ENGFRONTENDINFRA/pages/2012184577/Frontend+Data+Set+Filters) on filters:
- The only valid ones are `autocomplete`, `dateRange` and `selection`. 
- `number` and `text` are fossils of some old implementations.
- We recently merged `checkbox` and `radio` into `selection`.
- I removed all docs for custom filters, they were wrong for sure. Custom filters currently don't work, although there is some code for them. We need to review the API, add a sample, and update docs after that. That's a seperate task. For this PR, we can safely remove all mentions of custom filters. 

You can confirm props in [the components code](https://github.com/liferay/liferay-portal/tree/master/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/components/filters). Note that at some point in the future, we plan to merge `autocomplete` into `selection` filter.

Reply:
> number and text are fossils of some old implementations.

Jurassic Leftovers ®️ 😂 

 > We recently merged checkbox and radio into selection.

Oh, nice :)